File: DemoServer/AdditionalLanguages/go/Basics/deleteDocument/DeleteDocument.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    deleteDocument("companies/5-A")
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

//region Demo
func deleteDocument(documentId string) error {

    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()

    //region Step_1
    err = session.DeleteByID(documentId, "")
    if err != nil {
        return err
    }
    //endregion

    //region Step_2
    err = session.SaveChanges()
    if err != nil {
        return err
    }
    //endregion

    return nil
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/basics/deleteDocument/DeleteDocument.java
package net.ravendb.demo.basics.deleteDocument;

//region Usings
import net.ravendb.client.documents.session.IDocumentSession;
//endregion
import net.ravendb.demo.common.DocumentStoreHolder;

public class DeleteDocument {

    public void run(RunParams runParams) {
        String documentID = runParams.getDocumentId();

        //region Demo
        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
            //region Step_1
            session.delete(documentID);
            //endregion

            //region Step_2
            session.saveChanges();
            //endregion
        }
        //endregion
    }

    public static class RunParams {
        private String documentId;

        public String getDocumentId() {
            return documentId;
        }

        public void setDocumentId(String documentId) {
            documentId = documentId;
        }
    }
}
